src/NetMQ/ReceivingSocketExtensions.cs

```diff
@@ -393,13 +370,9 @@ public static void ReceiveMessage([NotNull] this IReceivingSocket socket, [NotNu
         {
             message.Clear();
 
-            bool more = true;
-
+            var more = true;
             while (more)
-            {
-                byte[] buffer = socket.Receive(dontWait, out more);
-                message.Append(buffer);
-            }
+                message.Append(socket.Receive(dontWait, out more));
```

Comment:
here we can call the method with the Msg parameter and improve performance a little by reusing the msg structure.